benchmarks/python/BUILD.bazel

```diff
@@ -0,0 +1,69 @@
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+load("@rules_python//python:defs.bzl", "py_library", "py_binary")
+load("@rules_pkg//:mappings.bzl", "pkg_files", "strip_prefix")
+load("//:protobuf.bzl", "py_proto_library")
+load("//benchmarks:defs.bzl", "benchmark_test")
+
+cc_binary(
+    name = "libbenchmark_messages.so",
+    srcs = ["python_benchmark_messages.cc"],
+    deps = [
+        "@local_config_python//:python_headers",
+        "//external:python_headers",
+        "//benchmarks:benchmarks_cc_proto",
+        "//benchmarks/datasets:cc_protos",
+    ],
+    linkshared = True,
+    linkstatic = True,
+    visibility = ["//visibility:private"],
+)
+
+py_binary(
+    name = "python_benchmark",
+    srcs = ["py_benchmark.py"],
+    main = "py_benchmark.py",
+    tags = ["benchmark"],
+    data = ["libbenchmark_messages.so"],
+    deps = [
+        "//:protobuf_python",
+        "//benchmarks:benchmarks_py_proto",
+        "//benchmarks/datasets:py_protos",
+    ],
+    env = select({
+        "//python:use_fast_cpp_protos": {"PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION": "cpp"},
+        "//conditions:default": {"PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION": "python"},
```

Comment:
I'm not sure that really fits the config setting model in Bazel.  If we add a //pyton:use_python_protos config_setting, the only way to make it mutually exclusive is to have it enabled on `--define=use_fast_cpp_protos=false`.  In this case there's no default and we either need to specify a default here anyway or error out when a user doesn't include a `--define=use_fast_cpp_protos`.

Am I misunderstanding something?

Reply:
Oh this might be my misunderstanding of how this works. Do users who want to use python just not use a flag?

If there is a flag that is used it might be nice to have that as an option and also have the default be python.

Reply:
Yea the way it works right now is that python is selected by default, but if you set `--define=use_fast_cpp_protos=true` then it uses the cpp implementation.  You could explicit specify `--define=use_fast_cpp_protos=false`, but it would just do the same thing as not setting the flag at all